analyzers/tests/SonarAnalyzer.UnitTest/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.TryCatchFinally.cs

```diff
@@ -113,10 +113,357 @@ public void Using_Nested_Declaration_LiveInUntilTheEnd()
+        {
+            var code = @"
+var usedAfter = 42;
+var usedInTry = 42;
+var usedInCatch = 42;
+try
+{
+    Method(usedInTry);
+}
+catch (Exception ex)
+{
+    Method(intParameter, usedInCatch, ex.HResult);
+}
+Method(usedAfter);";
+            var context = new Context(code);
+            context.ValidateEntry(new LiveIn("intParameter"), new LiveOut("intParameter"));
+            context.Validate("Method(usedInTry);", new LiveIn("usedInTry", "usedAfter", "usedInCatch", "intParameter"), new LiveOut("usedAfter", "usedInCatch", "intParameter"));
+            context.Validate("Method(intParameter, usedInCatch, ex.HResult);", new LiveIn("intParameter", "usedInCatch", "usedAfter"), new LiveOut("usedAfter"));
+            context.Validate("Method(usedAfter);", new LiveIn("usedAfter"));
+            context.ValidateExit();
+        }
+
+        [TestMethod]
+        public void Catch_SingleTypeWhenCondition_LiveIn()
+        {
+            var code = @"
+var usedAfter = 42;
+var usedInTry = 42;
+var usedInCatch = 42;
+try
+{
+    Method(usedInTry);
+}
+catch (Exception ex) when (ex.InnerException == null)
+{
+    Method(intParameter, usedInCatch, ex.HResult);
+}
+Method(usedAfter);";
+            var context = new Context(code);
+            context.ValidateEntry(new LiveIn("intParameter"), new LiveOut("intParameter"));
+            context.Validate("Method(usedInTry);", new LiveIn("usedInTry", "usedAfter", "usedInCatch", "intParameter"), new LiveOut("usedAfter", "usedInCatch", "intParameter"));
+            context.Validate("Method(intParameter, usedInCatch, ex.HResult);", new LiveIn("intParameter", "usedInCatch", "usedAfter", "ex"), new LiveOut("usedAfter"));
+            context.Validate("Method(usedAfter);", new LiveIn("usedAfter"));
+            context.ValidateExit();
+        }
+
+        [TestMethod]
+        public void Catch_SingleTypeWhenConditionReferencingArgument_LiveIn()
+        {
+            var code = @"
+try
+{
+    Method(0);
+}
+catch (Exception ex) when (boolParameter)
+{
+    Method(intParameter);
+}
+Method(1);";
+            var context = new Context(code);
+            context.ValidateEntry(new LiveIn("boolParameter", "intParameter"), new LiveOut("boolParameter", "intParameter"));
+            context.Validate("Method(0);", new LiveIn("boolParameter", "intParameter"), new LiveOut("boolParameter", "intParameter"));
+            context.Validate("Method(intParameter);", new LiveIn("intParameter"));
+            context.Validate("Method(1);");
+            context.ValidateExit();
+        }
+
+        [TestMethod]
+        public void Catch_MultipleTypes_LiveIn()
+        {
+            var code = @"
+var usedAfter = 42;
+var usedInTry = 42;
+var usedInCatchA = 42;
+var usedInCatchB = 42;
+try
+{
+    Method(usedInTry);
+}
+catch (FormatException ex)
+{
+    Method(intParameter, usedInCatchA, ex.HResult);
+}
+catch (Exception ex)
+{
+    Method(intParameter, usedInCatchB, ex.HResult);
```

Comment:
Can we add one more example when we catch exception and throw it without rethrow?

try
                {

                }
                catch (Exception ex)
                {
                    throw ex;
                }

I know that this shouldn't change anything, but this will still be worth testing here.